host/sgx/create.c

```diff
@@ -685,27 +685,27 @@ static oe_result_t _add_eeid_pages(
 
     if (eeid)
     {
+        char* eeid_bytes = (char*)eeid;
         size_t num_bytes = oe_eeid_byte_size(eeid);
         size_t num_pages =
             num_bytes / OE_PAGE_SIZE + ((num_bytes % OE_PAGE_SIZE) ? 1 : 0);
 
-        oe_page_t* pages =
-            oe_memalign(OE_PAGE_SIZE, sizeof(oe_page_t) * num_pages);
-        memset(pages, 0, sizeof(oe_page_t) * num_pages);
-        memcpy(pages->data, eeid, num_bytes);
-
+        oe_page_t* page = oe_memalign(OE_PAGE_SIZE, sizeof(oe_page_t));
         for (size_t i = 0; i < num_pages; i++)
         {
+            memset(page->data, 0, sizeof(oe_page_t));
+            size_t n = (i != num_pages - 1) ? OE_PAGE_SIZE
+                                            : (num_bytes % OE_PAGE_SIZE);
+            memcpy(page->data, eeid_bytes + OE_PAGE_SIZE * i, n);
+
             uint64_t addr = enclave_addr + *vaddr;
-            uint64_t src = (uint64_t)(pages + i * OE_PAGE_SIZE);
+            uint64_t src = (uint64_t)(page->data);
```

Comment:
Not part of this PR, but do eeid pages have to have `SGX_SECINFO_W` flag?

Reply:
Good observation - I don't think it's needed anymore now that we have the marker page. I'll run some tests to make sure.

Reply:
Done, I've added it to this PR too.